lib/galaxy/datatypes/tabular.py

```diff
@@ -309,13 +309,17 @@ def type_overrules_type(column_type1, column_type2):
 
         def is_int(column_text):
             try:
+                if any(c in ['_'] for c in column_text):
+                    return False
                 int(column_text)
                 return True
             except ValueError:
                 return False
 
         def is_float(column_text):
             try:
+                if any(c in ['_'] for c in column_text):
```

Comment:
```suggestion
                # Don't allow underscores in numeric literals (PEP 515)
                if '_' in column_text:
```

Reply:
or slightly cleaner, move the check out of the try/except